internal/app/singularity/registry.go

```diff
@@ -58,9 +58,16 @@ func (l *Library) Pull(ctx context.Context, from, to, arch string) error {
 		return fmt.Errorf("could not get image info: %v", err)
 	}
 
+	// don't use cached image
 	if l.cache.IsDisabled() {
-		// don't use cached image
-		err := l.pullAndVerify(ctx, imageMeta, libraryPath, to, arch)
+		// Since we are downloading it directly to the output image, we
+		// need to remove the overridden image if exists.
+		err := os.Remove(to)
```

Comment:
What if next pull returns an error? In that case will not get image pull but also remove the old one.